pentaho-osgi-utils/impl/src/main/java/org/pentaho/osgi/impl/KarafFeatureWatcherImpl.java

```diff
@@ -125,15 +125,19 @@ private void waitForFeatures( List<String> requiredFeatures ) throws Exception {
           IServiceBarrier serviceBarrier =
               IServiceBarrierManager.LOCATOR.getManager().getServiceBarrier( "KarafFeatureWatcherBarrier" );
           if ( serviceBarrier == null || serviceBarrier.isAvailable() ) {
-            logger.debug( getFeaturesReport( featuresService, uninstalledFeatures ) );
+            if ( logger.isDebugEnabled() ) {
```

Comment:
Just curious. Why do we have to check for if the log level is enabled. Won't the internal logic of the logger only log the line if the current log level is configured for it?

Reply:
That method call does a lot of work in there, and would get executed regardless of the log level.  It's more efficient to wrap this in the if statement so we don't waste time generating a debug message that'll never get seen.